pkg/proc/target.go

```diff
@@ -203,6 +207,21 @@ func (t *Target) IsCgo() bool {
 	return false
 }
 
+// Valid returns true if this Process can be used. When it returns false it
+// also returns an error describing why the Process is invalid (either
+// ErrProcessExited or ErrProcessDetached).
+func (t *Target) Valid() (bool, error) {
+	ok, err := t.proc.Valid()
+	if !ok && err != nil {
+		var pe ErrProcessExited
+		if errors.As(err, &pe) {
```

Comment:
`errors.As` has been introduced fairly recently, technically it's in all versions we support but every time we bump the minimum version required to compile delve we get complaints. Since it isn't that complicated to do this without `errors.As` (Valid doesn't chain ErrProcessExited with anything), I'd do it without it.

Reply:
Or, maybe golang.org/x/xerrors (intended for transition golang.org/design/29934-error-values)

Reply:
Done.